api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -352,16 +353,19 @@ public List<UserRole> getFirecloudUserRoles(String workspaceNamespace, String fi
     Map<String, RawlsWorkspaceAccessEntry> emailToRole =
         workspaceAuthService.getFirecloudWorkspaceAcl(workspaceNamespace, firecloudName);
 
+    var userMap = userDao.getUsersMappedByUsernames(emailToRole.keySet());
+
     List<UserRole> userRoles = new ArrayList<>();
     for (Map.Entry<String, RawlsWorkspaceAccessEntry> entry : emailToRole.entrySet()) {
```

Comment:
Could this for loop be on `userMap` instead of `emailToRole.entrySet()`? 

I think this would prevent the need for `DbUser user = userMap.get(entry.getKey());`, but I might be missing something.

Reply:
since userMap is also a Map, it's pretty much the same.